src/Listeners/CheckTags.php

```diff
@@ -49,19 +49,63 @@ public function handle(Saving $event)
             if (isset($event->data['relationships']['tags']['data'])) {
                 $tags = $event->data['relationships']['tags']['data'];
 
+                $allowedParentTagId = [];
+                $tagIdsProcessed = [];
+
                 foreach ($tags as $tag) {
                     $t = $this->getTagFromId($tag['id']);
 
+                    // Prepare for fancy primary + secondary tag interoperability!
+                    $tParent = null;
+                    $tParentId = null;
+
+                    // If the tag we're looking at right now is a secondary...
+                    if (null !== $t->parentId()) {
+                        $tParentId = $t->parentId();
+                        // Set its parent (primary) to the $tParent variable
+                        $tParent = $this->getTagFromId($tParentId);
+                    }
+
+                    // If this tag's slug doesn't match the forced tag for PDs...
                     if ($t->slug !== $this->byobuSlug) {
-                        throw new ValidationException(['byobu' => 'Invalid tag for private discussions']);
+                        // ...but this tag matches the allowed primary tag
+                        if (!empty($allowedParentTagId) && in_array($tag['id'], $allowedParentTagId, true)) {
+                            // It's all good in the hood!
+                        } else {
+                            // Tell ourselves that we've checked the tag with this ID
+                            // And it was validated to be INVALID
+                            $tagIdsProcessed += [($tag['id']) => false];
+                            continue;
+                        }
+                    } else {
+                        // If this allowed tag is a secondary tag...
+                        if (isset($tParentId)) {
+                            // Tell ourselves that we're allowed the primary too!
+                            array_push($allowedParentTagId, $tParentId);
+
+                            // If we've already processed the primary tag earlier, set it to be valid
+                            if (array_key_exists($tParentId, $tagIdsProcessed)) {
+                                $tagIdsProcessed[$tParentId] = true;
+                                continue;
+                            }
+                        }
                     }
+
+                    // Tell ourselves that we've checked the tag with this ID
+                    // And it was validated to be valid
+                    $tagIdsProcessed += [($tag['id']) => true];
+                }
+
+                if (in_array(false, $tagIdsProcessed, true)) {
+                    // A tag on the post was found to be invalid!
+                    throw new ValidationException(['byobu' => 'Invalid tag for private discussions (IDs: '.implode(array_keys($tagIdsProcessed, false, true), ',').')']);
```

Comment:
Bit of a syntax error going on here ;)

Reply:
You're going to have to point it out to me because my eyes just can't see it. I can't be getting old already...